services/cargohold/test/integration/TestSetup.hs

```diff
@@ -0,0 +1,31 @@
+module TestSetup
+    ( test
+    , tsManager
+    , tsCargohold
+    , TestSignature
+    , TestSetup(..)
+    , CargoHold
+    ) where
+
+import           Imports
+import           Bilge            (Request)
+import           Bilge.IO         (Http, Manager, runHttpT)
+import           Control.Lens     ((^.), makeLenses)
+import           Test.Tasty
+import           Test.Tasty.HUnit
+
+type CargoHold = Request -> Request
+type TestSignature a = CargoHold -> Http a
+
+data TestSetup = TestSetup
+  { _tsManager   :: Manager
+  , _tsCargohold :: CargoHold
+  }
+makeLenses ''TestSetup
+
+test :: IO TestSetup -> TestName -> (TestSignature a) -> TestTree
```

Comment:
too many parens in type signature.  but they are all over the place, feel free to ignore.

Reply:
Yeah; that's from massive search & replaces everywhere. I can probably use some regex magic to fix that :)